DTAdmin/Subjects/SubjectTableViewController.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  SubjectTableViewController.swift
+//  DTAdmin
+//
+//  Created by ITA student on 10/18/17.
+//  Copyright © 2017 if-ios-077. All rights reserved.
+//
+
+import UIKit
+
+class SubjectTableViewController: UITableViewController {
+
+    var records = [Records]()
+    
+    let queryService = QueryService()
+    
+    var filteredData = [Records]()
+    
+    var inSearchMode = false
+    
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        showRecords()
+        searchBar.showsScopeBar = true
+        searchBar.scopeButtonTitles = ["Name","Description"]
+        searchBar.selectedScopeButtonIndex = 0
+    }
+    
+    func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        if searchBar.text == nil || searchBar.text == "" {
+            inSearchMode = false
+            view.endEditing(true)
+            tableView.reloadData()
+        } else {
+            inSearchMode = true
+            switch searchBar.selectedScopeButtonIndex {
+            case 0:
+                filteredData = records.filter{$0.name.contains(searchBar.text!)}
+            case 1:
+                filteredData = records.filter{$0.description.contains(searchBar.text!)}
+            default:
+                print("No match")
+            }
+            tableView.reloadData()
+        }
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        showRecords()
+    }
+    
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+    }
+    
+    private func showRecords() {
+        Records.getRecords(sufix: "getRecords", completion: { (results:[Records]?) in
+            if let subjectData = results {
+                self.records = subjectData
+                self.records.sort { (rec1, rec2) -> Bool in
+                    return rec1.name < rec2.name
```

Comment:
`self.records.sort { return $0.name < $1.name }` is shorter -> IMO, better. Blood magic, Yay!